Automattic/wp-calypso
# A/B Test: Concierge Quickstart session upsell to users after plan purchase

#### Changes proposed in this Pull Request

* Cloned `ConciergeSessionNudge` component to `ConciergeQuickstartSession` for the new AB test with new copy text. The cloned component also includes some (minor) design tweaks (headers, removed overflow:hidden, etc).

#### Testing instructions

Start Calypso locally with this branch and visit http://calypso.localhost:3000/.

1. Put yourself into the `variantQuickstartSession` test group for `conciergeQuickstartSession`.

![Screen Shot on 2019-04-10 at 13:02:19](https://user-images.githubusercontent.com/538790/55875361-34775000-5b95-11e9-9d20-a9422a2b7122.png)

2. Create a new site (to make sure the last redirect is consistent, **don't** choose "_Sell products or collect payments_" for the "_What’s the primary goal..._" question) or use an existing test site.

3. Purchase or upgrade to a plan (no domain).

4. After purchase, confirm you're shown the Concierge upsell page with the `Quick Start` copy.

![image](https://user-images.githubusercontent.com/538790/56048046-f3bf3880-5d46-11e9-92ad-cf4983401b4d.png)

5. Click the button: "Yes, I want a WordPress Expert by my side". You should be taken to the checkout with the session added to the cart.

<img width="292" alt="Screen Shot 2019-04-10 at 13 03 25" src="https://user-images.githubusercontent.com/538790/55875499-861fda80-5b95-11e9-9ed3-39d0b8e1ebf7.png">

6. Purchase the session.

7. Confirm you're either taken to the checklist or to the Thank you page.

<img width="851" alt="screen shot 2019-01-02 at 11 21 46 am" src="https://user-images.githubusercontent.com/690843/50608428-81a2a200-0e81-11e9-9234-07da29eb5c98.png">

<img width="1009" alt="Screen Shot 2019-04-10 at 13 38 59" src="https://user-images.githubusercontent.com/538790/55875726-15c58900-5b96-11e9-8004-8fe0bbe9fe80.png">


-------------

Next, put yourself into the `controlSupportSession` test group and run through steps 2-4 again.
This time you should be shown the Concierge upsell page with the control copy:

<img width="764" alt="Screen Shot 2019-04-10 at 13 02 57" src="https://user-images.githubusercontent.com/538790/55875587-b23b5b80-5b95-11e9-87f0-7cce52b0c9a1.png">


#### Comments

I'm just now realizing I should've committed the cloned component earlier, so real changes would have shown up in the diff, instead of the whole file. Apologies!

Here are the diffs between `ConciergeSessionNudge` files (left) and `ConciergeQuickstartSession` (right): 

* [index.jsx](https://www.diffchecker.com/nSynBbd8)
* [style.scss](https://www.diffchecker.com/J1SgLHb6)

#### Questions

* ~~How were translations/audiences dealt with in the previous test? Are we running this AB test only to English audiences?~~ Just answered this by actually paying attention to the [AB test README](https://github.com/Automattic/wp-calypso/tree/master/client/lib/abtest). Short answer: By default, tests only run on users where the locale is set to English.
* This test is supposed to exclude the GSuite version of the flow. I've applied the `abtest()` logic [here](https://github.com/Automattic/wp-calypso/pull/32176/files#diff-199e3967ae413c2ff48db966abbf8386) for the non-eligible for GSuite flow, which should be ok, but would appreciate another pair of eyes here.
* Also related: can't help but noticed that [there's an abtest() call](https://github.com/Automattic/wp-calypso/blob/master/client/my-sites/checkout/checkout/index.jsx#L420) which looks like leftovers from the previous Concierge test for GSuite. Is this ok to remove?
* Noob question: after the review process, is deploying these changes enough to have the AB test already running? or is there anything else missing?
